recipes/uchardet/all/conanfile.py

```diff
@@ -0,0 +1,92 @@
+from conans import ConanFile, CMake, tools
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class UchardetConan(ConanFile):
+    name = "uchardet"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/freedesktop/uchardet"
+    description = "uchardet is an encoding detector library, which takes a sequence of bytes in an unknown character encoding and attempts to determine the encoding of the text. Returned encoding names are iconv-compatible."
+    topics = "encoding", "detector"
+    license = "MPL 1.1"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "check_sse2": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "check_sse2": True,
+    }
+
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
```

Comment:
`sse2` does notmake sense on e.g. arm.
```suggestion
    def config_options(self):
        if self.settings.arch not in ("x86", "x86_64"):
            del self.options.check_sse2
        if self.settings.os == "Windows":
```